src/store/printer/getters.ts

```diff
@@ -111,25 +111,48 @@ export const getters: GetterTree<PrinterState, RootState> = {
       if (currentPosition > 0 && endPosition > 0) return currentPosition / endPosition
     }
 
-    return state.printer.display_status.progress || 0
+    return progress || 0
+  },
+
+  getFileAbsolutePrintProgress: (state): number => {
+    return state.printer.virtual_sdcard?.progress || 0
   },
 
   getSlicerPrintProgress: (state): number => {
     return state.printer.display_status.progress || 0
   },
 
+  getFilamentPrintProgress: (state) => {
+    const { filament_used, filename: statsFilename } = state.printer.print_stats ?? {}
+    const { filament_total, path, filename } = state.printer.current_file ?? {}
+
+    const fullFilename = path ? `${path}/${filename}` : filename
+
+    if (filament_used && filament_total && fullFilename === statsFilename) {
```

Comment:
I think the `filament_used` check here should be a nullish check instead of a falsy check. Right now, when starting a print, the progress starts out at 5% for me (fallback to virtual_sdcard.progress), and then jumps to 1% as soon as filament is extruded.
Otherwise LGTM!

Reply:
Ah, very nice find, I've added the check!